### PR TITLE
Add `name` to `wp_register_webfonts`

### PR DESCRIPTION
**Pull request by simison**
## What?
Solves https://github.com/WordPress/gutenberg/issues/46398

Adds support for `name` attribute in `wp_register_webfonts`:

```php
array(
	'name'         => 'Example (Japanese)',
	'font-family'  => 'Example',

	'font-style'   => 'normal',
	'font-weight'  => '200 900',
	'font-stretch' => 'normal',
	'src'          => 'https://example.com/assets/fonts/source-serif-pro/SourceSerif4Variable-Roman.ttf.woff2',
	'font-display' => 'fallback',
	'provider'     => 'local',
)
```

## Why?
I'm describing use case in detail at https://github.com/WordPress/gutenberg/issues/46398, but essentially to match how `theme.json` already works.

## How?
Allows passing `name` attribute as part of font family array.

I'm not entirely happy with solution as conceptually "Name" would belong to higher up, not necessarily inside font-family definition. I also didn't see better alternative.

## Testing Instructions

Add this to mu-plugins or elsewhere:

```php
function register_example_font() {
	$font_family = "Example";
	$font_name = "Example (Japanese)";
	
	wp_register_webfonts(
		array(
			array(
				'provider'     => 'local',
				'name'         => $font_name,
				'font-family'  => $font_family,
				'font-style'   => 'normal',
				'font-weight'  => '200 900',
				'font-stretch' => 'normal',
				'src'          => 'https://example.com/assets/fonts/source-serif-pro/SourceSerif4Variable-Roman.ttf.woff2',
				'font-display' => 'fallback',
			),
			array(
				'provider'     => 'local',
				'name'         => $font_name,
				'font-family'  => $font_family,
				'font-style'   => 'italic',
				'font-weight'  => '200 900',
				'font-stretch' => 'normal',
				'src'          => 'https://example.com/assets/fonts/source-serif-pro/SourceSerif4Variable-Italic.ttf.woff2',
				'font-display' => 'fallback',
			),
		)
	);
}
add_filter( 'init', 'register_example_font' );
```

Open site editor and note font families:

Screenshot 2022-12-08 at 15 11 40

### CODE REVIEW

File: lib/experimental/register-webfonts-from-theme-json.php
Comment (spacedmonkey) [inline lib/experimental/register-webfonts-from-theme-json.php]: Alignment is wrong.

Comment (simison) [reply lib/experimental/register-webfonts-from-theme-json.php]: thanks, [fixed](https://github.com/WordPress/gutenberg/pull/46404/commits/17983a554d65dbe0fb1586c23555a1eac52ee285)

**Review by hellofromtonya — CHANGES_REQUESTED** [review]
Block this PR until the architectural redesign is completed and merged. (See my comment here https://github.com/WordPress/gutenberg/issues/46398#issuecomment-1342965780 for more context.) Once merged, then can revisit what changes are needed in this PR, if any.

**Comment by hellofromtonya** [comment]
This PR is now unblocked. It will need to be rebased and rebuilt to apply the enhancement to the newly designed Fonts API (which is now in `trunk`). Please note, the API has a brand new architecture and was renamed (including all of the files, functions, classes, tests, etc.).

**Comment by hellofromtonya** [comment]
This PR was applied to the old API. It will not work with the new API. I think it's best to close this PR, but use it as reference for the intent. Then open a new PR to add the enhancement to the new API.

**Pull request closed without merging**
